aws/lambda/clickhouse-replicator-s3/lambda_function.py

```diff
@@ -0,0 +1,409 @@
+        result,
+        {get_skipped_failure_parser_helper('skipped', 'Tuple(type String, message String, text String)', 'type')},
+        status,
+        {get_sys_err_out_parser('system-err')},
+        {get_sys_err_out_parser('system-out')},
+        time,
+        now()::DateTime64(9) as time_inserted,
+        type_param,
+        value_param,
+        workflow_id,
+        workflow_run_attempt,
+        ('{bucket}', '{key}')
+    from
+        s3(
+            'https://{bucket}.s3.amazonaws.com/{encode_url_component(key)}',
+            'JSONEachRow',
+            '
+            `classname` String,
+            `duration` Float32,
+            `error` String,
+            `failure` String,
+            `file` String,
+            `invoking_file` String,
+            `job_id` Int64,
+            `line` Float32,
+            `name` String,
+            `properties` Tuple(property Tuple(name String, value String)),
+            `rerun` String,
+            `result` String,
+            `skipped` String,
+            `status` String,
+            `system-err` String,
+            `system-out` String,
+            `time` Float32,
+            `type_param` String,
+            `value_param` String,
+            `workflow_id` Int64,
+            `workflow_run_attempt` Int32',
+            'gzip'
+        )
+    """
+    query = clean_up_query(query)
+    try:
+        get_clickhouse_client().query(query)
+    except Exception as e:
+        if "Expected not greater than" in str(e):
+            get_clickhouse_client().query(
+                f"insert into errors.{table}_ingest_errors values ('{key}', 'file is too large?')"
+            )
+        else:
+            get_clickhouse_client().query(
+                f"insert into errors.{table}_ingest_errors values ('{key}', '{json.dumps(str(e))}')"
+            )
+
+
+def rerun_disabled_tests_adapter(table, bucket, key):
+    schema = """
+    `classname` String,
+    `filename` String,
+    `flaky` Bool,
+    `name` String,
+    `num_green` Int64,
+    `num_red` Int64,
+    `workflow_id` Int64,
+    `workflow_run_attempt` Int64
+    """
+
+    url = f"https://{bucket}.s3.amazonaws.com/{encode_url_component(key)}"
+
+    def get_insert_query(compression):
+        return f"""
+        insert into {table}
+        select *, ('{bucket}', '{key}')
+        from s3('{url}', 'JSONEachRow', '{schema}', '{compression}')
+        """
+
+    try:
+        get_clickhouse_client().query(get_insert_query("gzip"))
+    except Exception as e:
+        if "Expected not greater than" in str(e):
+            get_clickhouse_client().query(
+                f"insert into errors.{table}_ingest_errors values ('{key}', 'file is too large?')"
+            )
+        else:
+            get_clickhouse_client().query(
+                f"insert into errors.{table}_ingest_errors values ('{key}', '{json.dumps(str(e))}')"
+            )
+
```

Comment:
A lot of adapters in order to handle different schemas.  They can be generalized, but I would prefer to do that at a later date